distributed/worker_state_machine.py

```diff
@@ -1888,6 +1888,9 @@ def _transition_from_resumed(
             assert next_state in {"waiting", "fetch"}, next_state
             assert ts._previous in {"executing", "flight"}, ts._previous
 
+            if ts._previous == "executing":
```

Comment:
TODO: Write test that ensures that we do in fact release IFF the task is finished.

Reply:
This and line 1888 are missing the use case of 'long-running'. I'll open a follow-up PR about it, no point adding scope creep here.

I'd like to have the test you mention within this PR please.

Reply:
`ts._previous` should never happen to begin with in this method. I'll clean up the code in here in a separate PR.